app/parse.py

```diff
@@ -8,5 +15,40 @@
     duration: str
 
 
-def get_all_courses() -> list[Course]:
-    pass
+COURSE_FIELDS = [field.name for field in fields(Course)]
+
+
+def parse_single_course(page_soup: BeautifulSoup) -> Course:
+    return Course(name=page_soup.select_one("h3").text,
+                  short_description=page_soup.select_one(
+                      ".typography_landingTextMain__Rc8BD.mb-32"
+                  ).text,
+                  duration=page_soup.select_one(
+                      "p.typography_landingTextMain__Rc8BD.ProfessionCardTags_regularTag__yTc6K:last-of-type > span:last-of-type"
+                  ).text
+                  )
```

Comment:
Add error handling for cases where the expected HTML elements are not found. You can use a try-except block or check if the elements are `None` before accessing their `text` attribute to avoid `AttributeError`.